diff --git a/package.json b/package.json
index 9fe17e7..54a6ccc 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -34,6 +34,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.11.0"
+    "initVersion": "6.11.4"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index e7b9d82..ac65443 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -24,6 +24,7 @@ const Home: NextPage = () => {
             <Link
               className={styles.card}
               href="https://create.t3.gg/en/usage/first-steps"
+              target="_blank"
             >
               <h3 className={styles.cardTitle}>First Steps →</h3>
               <div className={styles.cardText}>
@@ -34,6 +35,7 @@ const Home: NextPage = () => {
             <Link
               className={styles.card}
               href="https://create.t3.gg/en/introduction"
+              target="_blank"
             >
               <h3 className={styles.cardTitle}>Documentation →</h3>
               <div className={styles.cardText}>
