diff --git a/package.json b/package.json
index 0fc03e5..f3533a8 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -36,6 +36,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.11.0"
+    "initVersion": "6.11.4"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index ccab419..1967114 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -23,6 +23,7 @@ const Home: NextPage = () => {
             <Link
               className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 text-white hover:bg-white/20"
               href="https://create.t3.gg/en/usage/first-steps"
+              target="_blank"
             >
               <h3 className="text-2xl font-bold">First Steps →</h3>
               <div className="text-lg">
@@ -33,6 +34,7 @@ const Home: NextPage = () => {
             <Link
               className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 text-white hover:bg-white/20"
               href="https://create.t3.gg/en/introduction"
+              target="_blank"
             >
               <h3 className="text-2xl font-bold">Documentation →</h3>
               <div className="text-lg">
