diff --git a/.env-example b/.env-example
index 3451e8e..b0c4cad 100644
--- a/.env-example
+++ b/.env-example
@@ -9,7 +9,7 @@
 # Next Auth
 # You can generate the secret via 'openssl rand -base64 32' on Linux
 # More info: https://next-auth.js.org/configuration/options#secret
-NEXTAUTH_SECRET=
+# NEXTAUTH_SECRET=
 NEXTAUTH_URL=http://localhost:3000
 
 # Next Auth Discord Provider
diff --git a/package.json b/package.json
index 2e44d93..c479abc 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -26,6 +26,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.8.0"
+    "initVersion": "6.8.1"
   }
 }
diff --git a/src/env/schema.mjs b/src/env/schema.mjs
index 9e17f4d..1a4d09f 100644
--- a/src/env/schema.mjs
+++ b/src/env/schema.mjs
@@ -7,7 +7,10 @@ import { z } from "zod";
  */
 export const serverSchema = z.object({
   NODE_ENV: z.enum(["development", "test", "production"]),
-  NEXTAUTH_SECRET: z.string(),
+  NEXTAUTH_SECRET:
+    process.env.NODE_ENV === "production"
+      ? z.string().min(1)
+      : z.string().min(1).optional(),
   NEXTAUTH_URL: z.preprocess(
     // This makes Vercel deployments not fail if you don't set NEXTAUTH_URL
     // Since NextAuth automatically uses the VERCEL_URL if present.
