diff --git a/package.json b/package.json
index 0932023..99fd809 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -28,6 +28,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.11.1"
+    "initVersion": "6.11.2"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index cbfff1a..35f4e30 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -20,6 +20,7 @@ const Home: NextPage = () => {
             <Link
               className={styles.card}
               href="https://create.t3.gg/en/usage/first-steps"
+              target="_blank"
             >
               <h3 className={styles.cardTitle}>First Steps →</h3>
               <div className={styles.cardText}>
@@ -30,6 +31,7 @@ const Home: NextPage = () => {
             <Link
               className={styles.card}
               href="https://create.t3.gg/en/introduction"
+              target="_blank"
             >
               <h3 className={styles.cardTitle}>Documentation →</h3>
               <div className={styles.cardText}>
