diff --git a/package.json b/package.json
index c41febb..b3a2c46 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -10,10 +10,10 @@
     "start": "next start"
   },
   "dependencies": {
-    "@next-auth/prisma-adapter": "1.0.5",
+    "@next-auth/prisma-adapter": "^1.0.5",
     "@prisma/client": "^4.5.0",
     "next": "13.0.2",
-    "next-auth": "4.17.0",
+    "next-auth": "^4.18.3",
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "zod": "^3.18.0"
@@ -35,6 +35,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.11.1"
+    "initVersion": "6.11.3"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 7e880c8..5513c42 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -19,6 +19,7 @@ const Home: NextPage = () => {
             <Link
               className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 text-white hover:bg-white/20"
               href="https://create.t3.gg/en/usage/first-steps"
+              target="_blank"
             >
               <h3 className="text-2xl font-bold">First Steps →</h3>
               <div className="text-lg">
@@ -29,6 +30,7 @@ const Home: NextPage = () => {
             <Link
               className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 text-white hover:bg-white/20"
               href="https://create.t3.gg/en/introduction"
+              target="_blank"
             >
               <h3 className="text-2xl font-bold">Documentation →</h3>
               <div className="text-lg">
