diff --git a/next.config.mjs b/next.config.mjs
index d921057..654e725 100644
--- a/next.config.mjs
+++ b/next.config.mjs
@@ -19,4 +19,5 @@ const config = {
     defaultLocale: "en",
   },
 };
+
 export default config;
diff --git a/package.json b/package.json
index ca2c956..995882e 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -36,6 +36,6 @@
     "typescript": "^5.0.4"
   },
   "ct3aMetadata": {
-    "initVersion": "7.13.1"
+    "initVersion": "7.14.0"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 3768eb3..2cda92c 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -1,10 +1,9 @@
 import styles from "./index.module.css";
-import { type NextPage } from "next";
 import Head from "next/head";
 import Link from "next/link";
 import { api } from "~/utils/api";
 
-const Home: NextPage = () => {
+export default function Home() {
   const hello = api.example.hello.useQuery({ text: "from tRPC" });
 
   return (
@@ -50,6 +49,4 @@ const Home: NextPage = () => {
       </main>
     </>
   );
-};
-
-export default Home;
+}
