diff --git a/.env.example b/.env.example
new file mode 100644
index 0000000..38b70c1
--- /dev/null
+++ b/.env.example
@@ -0,0 +1,11 @@
+# Since .env is gitignored, you can use .env.example to build a new `.env` file when you clone the repo.
+# Keep this file up-to-date when you add new variables to `.env`.
+
+# This file will be committed to version control, so make sure not to have any secrets in it.
+# If you are cloning this repo, create a copy of this file named `.env` and populate it with your secrets.
+
+# When adding additional env variables, the schema in /env/schema.mjs should be updated accordingly
+
+# Example:
+# SERVERVAR=foo
+# NEXT_PUBLIC_CLIENTVAR=bar
diff --git a/package.json b/package.json
index b431e46..cf45c9c 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -25,6 +25,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.10.3"
+    "initVersion": "6.11.1"
   }
 }
