diff --git a/.gitignore b/.gitignore
index 2971a0b..9b986df 100644
--- a/.gitignore
+++ b/.gitignore
@@ -11,6 +11,7 @@
 # database
 /prisma/db.sqlite
 /prisma/db.sqlite-journal
+db.sqlite
 
 # next.js
 /.next/
diff --git a/next.config.js b/next.config.js
index ffbeb9f..98b6f90 100644
--- a/next.config.js
+++ b/next.config.js
@@ -17,6 +17,7 @@ const config = {
     locales: ["en"],
     defaultLocale: "en",
   },
+  transpilePackages: ["geist"],
 };
 
 export default config;
diff --git a/package.json b/package.json
index e2adf89..58c8c01 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "type": "module",
@@ -10,11 +10,12 @@
     "start": "next start"
   },
   "dependencies": {
-    "@t3-oss/env-nextjs": "^0.9.2",
+    "@t3-oss/env-nextjs": "^0.10.1",
+    "geist": "^1.3.0",
     "next": "^14.2.1",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "zod": "3.22.4"
+    "zod": "^3.23.3"
   },
   "devDependencies": {
     "@types/eslint": "^8.56.2",
@@ -32,7 +33,7 @@
     "typescript": "^5.4.2"
   },
   "ct3aMetadata": {
-    "initVersion": "7.31.0"
+    "initVersion": "7.33.1"
   },
   "packageManager": "pnpm@8.5.1"
 }
diff --git a/src/pages/_app.tsx b/src/pages/_app.tsx
index 54fd82a..e9d159b 100644
--- a/src/pages/_app.tsx
+++ b/src/pages/_app.tsx
@@ -1,16 +1,11 @@
+import { GeistSans } from "geist/font/sans";
 import { type AppType } from "next/app";
-import { Inter } from "next/font/google";
 
 import "~/styles/globals.css";
 
-const inter = Inter({
-  subsets: ["latin"],
-  variable: "--font-sans",
-});
-
 const MyApp: AppType = ({ Component, pageProps }) => {
   return (
-    <main className={`font-sans ${inter.variable}`}>
+    <main className={GeistSans.className}>
       <Component {...pageProps} />
     </main>
   );
diff --git a/tailwind.config.ts b/tailwind.config.ts
index f06488f..5fd44e8 100644
--- a/tailwind.config.ts
+++ b/tailwind.config.ts
@@ -6,7 +6,7 @@ export default {
   theme: {
     extend: {
       fontFamily: {
-        sans: ["var(--font-sans)", ...fontFamily.sans],
+        sans: ["var(--font-geist-sans)", ...fontFamily.sans],
       },
     },
   },
