diff --git a/.gitignore b/.gitignore
index d9bbc5d..9dfddc1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -15,6 +15,7 @@
 # next.js
 /.next/
 /out/
+next-env.d.ts
 
 # production
 /build
diff --git a/package.json b/package.json
index 9ea4884..db7355a 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -9,11 +9,11 @@
     "start": "next start"
   },
   "dependencies": {
-    "@tanstack/react-query": "^4.10.0",
-    "@trpc/client": "10.0.0-rc.4",
-    "@trpc/next": "10.0.0-rc.4",
-    "@trpc/react-query": "10.0.0-rc.4",
-    "@trpc/server": "10.0.0-rc.4",
+    "@tanstack/react-query": "^4.16.0",
+    "@trpc/client": "^10.0.0",
+    "@trpc/next": "^10.0.0",
+    "@trpc/react-query": "^10.0.0",
+    "@trpc/server": "^10.0.0",
     "next": "13.0.2",
     "react": "18.2.0",
     "react-dom": "18.2.0",
@@ -31,6 +31,6 @@
     "typescript": "^4.8.4"
   },
   "ct3aMetadata": {
-    "initVersion": "6.10.0"
+    "initVersion": "6.10.2"
   }
 }
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 357daed..e7b9d82 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -21,14 +21,20 @@ const Home: NextPage = () => {
             Create <span className={styles.pinkSpan}>T3</span> App
           </h1>
           <div className={styles.cardRow}>
-            <Link className={styles.card} href="https://beta.nextjs.com">
+            <Link
+              className={styles.card}
+              href="https://create.t3.gg/en/usage/first-steps"
+            >
               <h3 className={styles.cardTitle}>First Steps →</h3>
               <div className={styles.cardText}>
                 Just the basics - Everything you need to know to set up your
                 database and authentication.
               </div>
             </Link>
-            <Link className={styles.card} href="https://beta.nextjs.com">
+            <Link
+              className={styles.card}
+              href="https://create.t3.gg/en/introduction"
+            >
               <h3 className={styles.cardTitle}>Documentation →</h3>
               <div className={styles.cardText}>
                 Learn more about Create T3 App, the libraries it uses, and how
